src/lib.rs

```diff
@@ -378,23 +439,42 @@ pub fn load_product(data: &[u8]) -> Result<&Product, PythError> {
 }
 
 /** Get a `Price` account from the raw byte value of a Solana account. */
-pub fn load_price(data: &[u8]) -> Result<&Price, PythError> {
-  let pyth_price = load::<Price>(&data).map_err(|_| PythError::InvalidAccountData)?;
+pub fn load_price_account_data(data: &[u8]) -> Result<&PriceAccountData, PythError> {
+  let price_account_data = load::<PriceAccountData>(&data).map_err(|_| PythError::InvalidAccountData)?;
 
-  if pyth_price.magic != MAGIC {
+  if price_account_data.magic != MAGIC {
     return Err(PythError::InvalidAccountData);
   }
-  if pyth_price.ver != VERSION_2 {
+  if price_account_data.ver != VERSION_2 {
     return Err(PythError::BadVersionNumber);
   }
-  if pyth_price.atype != AccountType::Price as u32 {
+  if price_account_data.atype != AccountType::Price as u32 {
     return Err(PythError::WrongAccountType);
   }
 
+  return Ok(price_account_data);
+}
+
+/** Get a modified `Price` struct from the raw byte value of a Solana Price account.
+ *  If used on-chain it will update the status to unknown if price is not updated for a long time.
+*/
+pub fn load_price(data: &[u8]) -> Result<Price, PythError> {
+  let price_account_data = load_price_account_data(data)?;
+
+  #[allow(unused_mut)]
+  let mut pyth_price = Price::from_price_account_data(price_account_data);
+
+  #[cfg(target_arch = "bpf")]
+  if let PriceStatus::Trading = pyth_price.agg.status {
+    if Clock::get().unwrap().slot - pyth_price.agg.pub_slot > MAX_SEND_LATENCY {
+      pyth_price.agg.status = PriceStatus::Unknown;
```

Comment:
I think people will find mutation here to be surprising. I think it would be better to make the existing accessor `get_current_price` implement the necessary freshness check.